txmanager.go

```diff
@@ -55,8 +56,27 @@ func (g *GormTxManager) WithTransaction(parentCtx context.Context, txfn TxFn) (e
 		}
 	}()
 
-	err = txfn(txCtx)
-	return err
+	//execute txfn in background and communicates errors through a channel
+	errCh := make(chan error)
+	go func() {
+		var errFunc error
+		defer func() {
+			if p := recover(); p != nil {
+				logrus.Errorf("stack trace: %s", g.printStackTrace())
+				errFunc = fmt.Errorf("panic recovered: %v", p)
+			}
+			errCh <- errFunc
+		}()
+		errFunc = txfn(txCtx)
+	}()
+
+	//wait for context cancelled or fn is success(err=nil) or failed(err=!nil)
+	select {
+	case <-parentCtx.Done():
+		return parentCtx.Err()
+	case err := <-errCh:
+		return err
+	}
 }
 
 func (g *GormV2TxManager) WithTransaction(parentCtx context.Context, txfn TxFn) (err error) {
```

Comment:
@bintangsibarani please also implement the rollback-on-context-canceled in the V2 version. Thanks